test/integration/resource.ts

```diff
@@ -0,0 +1,43 @@
+import {expect} from 'chai';
+import {Ketting, NeverCache} from '../../src';
+
+describe('Using resources', async () => {
+  const ketting = new Ketting('http://localhost:3000/hal1.json');
+
+  it('should do 2 distinct HTTP calls if request headers are different on concurrent calls', async () => {
+    ketting.cache = new NeverCache();
+
+    const resource = await ketting.follow('collection');
+
+    const [firstState, secondState] =
+      await Promise.all([
+        resource.get({headers: {'delay-in-ms': '2', 'prefer': 'bar'}}),
+        resource.get({headers: {'delay-in-ms': '2', 'prefer': 'baz'}})
+      ]);
+
+    expect(firstState.headers.get('preference-applied')).to.not.eql(secondState.headers.get('preference-applied'));
+  });
+
+  it('should produce only one HTTP call if request headers are the same on concurrent calls', async () => {
+    ketting.cache = new NeverCache();
+
+    const resource = await ketting.follow('collection');
+
+    const headers1 = new Headers();
+    headers1.append('prefer', 'bar');
```

Comment:
`prefer` is a standard header. `bar` is not a valid value for `prefer`.
Use a fake header name (as you did before) to keep the test clean.

Reply:
Now that we no longer use `prefer` header, removing `preference-applied` response header.

Reply:
> Use a fake header name (as you did before) to keep the test clean.
Sorry, wrong conclusion from me. Use a valid `prefer` value please.

Reply:
done